hyperspace/primitives/src/lib.rs

```diff
@@ -366,6 +384,18 @@ pub trait Chain: IbcProvider + KeyProvider + Send + Sync {
 		&self,
 		messages: Vec<Any>,
 	) -> Result<(sp_core::H256, Option<sp_core::H256>), Self::Error>;
+
+	/// Returns an [`AnyClientMessage`] that triggered an [`IbcEvent`] at index `event_index`
+	/// on this chain.
+	///
+	/// Parameters `host_block_hash` and `transaction_index` identify the transaction that triggered
+	/// the event.
+	async fn query_client_message(
+		&self,
+		host_block_hash: [u8; 32],
```

Comment:
I think we can encapsulate these parameters in an enum with a variant for Parachains, because these current parameters seem parachain specific, when other chains are supported, the enum can be modified with variants for those new chains.

Reply:
Yes, this will be changed, Seun will suggest some solution